lib/models/device.coffee

```diff
@@ -249,6 +249,41 @@ exports.getApplicationName = (uuid, callback) ->
 	exports.get(uuid).get('application_name').nodeify(callback)
 
 ###*
+# @summary Get application container information
+# @name getApplicationInfo
+# @public
+# @function
+# @memberof resin.models.device
+#
+# @param {String} uuid - device uuid
+# @fulfil {Object} - application info
+# @returns {Promise}
+#
+# @example
+# resin.models.device.getApplicationInfo('7cf02a6').then(function(appInfo) {
+# 	console.log(appInfo);
+# });
+#
+# @example
+# resin.models.device.getApplicationInfo('7cf02a6', function(error, appInfo) {
+# 	if (error) throw error;
+# 	console.log(appInfo);
+# });
+###
+exports.getApplicationInfo = (uuid, callback) ->
+	exports.get(uuid).then (device) ->
```

Comment:
I've already added a supervisor check, let's wait for @Page- and I can happily revert the commit if we eventually decide not to have a check on the SDK. 

Imo it makes sense to me to have the API supervisor proxy as simple as possible, which I would expect to transparently forward requests to the device.

The supervisor API is, after all, documented and the SDK has the purpose of facilitating its use, so I wouldn't mind adding supervisor version checks here (and to other SDKs, like the python one). Users who choose to directly use the API (e.g. with curl) aren't protected from making a wrong API call anyway. - By the way, how does the Supervisor API proxy handle non-existent endpoints?